pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -178,102 +176,99 @@ type CertificateRequestPolicyAllowedX509Subject struct {
 	SerialNumber *CertificateRequestPolicyAllowedString `json:"serialNumber,omitempty"`
 }
 
-// CertificateRequestPolicyAllowedStringSlice represents an allowed string
-// slice value paired with whether the field is a required value on the
-// request.
+// CertificateRequestPolicyAllowedStringSlice represents allowed string values paired
+// with whether the field is a required value on the request.
+// An empty Values paired with Required `true` is an impossible condition that
```

Comment:
Hmm... maybe we should talk about `impossible`.

Is the idea that "you can't satisfy this requirement and thus this Policy will not apply, and thus unless some other policy does, your request will not be fulfilled"?

Reply:
If we could use "`Values`" instead of "Values", it'd be really helpful for readers, as it's really confusing -- as a reader, I expect to see `An empty value paired...` but what the text wants to do is talk about the `Values` field which is a different creature entirely.

```suggestion
// `Values: []` paired with `Required: true` establishes a policy that will never grant a `CertificateRequest`, but other policies may.
```

The reason I am proposing `will not grant` is that `deny` implies a final outcome, but that isn't what happens in cert-manager -- a `CertificateRequest` can be processed by any other `CertificateRequestPolicy`s and a `Certificate` issued by any that do _allow_ the `CertificateRequest`.

Reply:
I am torn here. The suggestion from Josh is a bit long, and I am also in doubt about the suggested field value notation. WDYT @SgtCoDFish @inteon? Impossible condition, "always-false" or the longer alternative? Somehow I don't understand why any user would create a policy that won't grant any request. This is hard, and I need to stop working on this PR soon. 😅

Reply:
Ok, I have now made some (hopefully) "final" adjustments - based on the suggestion from @jsoref. I am hoping we are approaching the finish line of this marathon now. PTAL!